File: pa3/pa3/MatrixTest.java
/*
    Zachary Dutra
    zdutra
    1581789
    pa3
*/
class MatrixTest{
    public static void main(String [] args){
        Matrix A = new Matrix(10);
        Matrix B = new Matrix(15);
        A.changeEntry(1, 1, 1);
        B.changeEntry(1, 1, 1);
        System.out.println(A.equals(B));
        B = new Matrix(10);
        A.changeEntry(1, 1, 1);
        A.changeEntry(1, 3, 1);
        B.changeEntry(1, 1, 1);
        B.changeEntry(1, 3, 1);
        System.out.print(A.toString());
        System.out.print(B.toString());
        System.out.println(A.equals(B));
        System.out.println(!A.equals(B));
        A.changeEntry(1, 3, 0);
        System.out.println(A.equals(B));
        A.changeEntry(1, 1, 0);
        B.makeZero();
        A.changeEntry(10, 10, 10);
        B.changeEntry(10, 10, 10);
        System.out.println(!A.equals(B));
        A = new Matrix(100);
        B = new Matrix(100);
        int valcount = 1;
        for (int j = 1; j <= 100; j++) {
          for (int k = 1; k <= 100; k++) {
            // hint: this is 1-10000 left-to-right row-by-row
            A.changeEntry(j, k, valcount++);
          }
          B.changeEntry(j, j, 1); // hint: this is the identity matrix
        }
        Matrix C = A.scalarMult(2);
        System.out.println(!C.equals(A.add(A)));
        C = A.scalarMult(-2);
        System.out.println(!C.equals(A.sub(A).sub(A).sub(A)));
        C = A.mult(B);
        System.out.println(!C.equals(A));
    }
}

File: CSE101/pa2/MatrixTest.c
/*
Zachary Dutra
zdutra
1581789
pa2
*/ 
#include<stdio.h>
#include<stdlib.h>
#include"List.h"
#include"Matrix.h"
int main(int argc, char* argv[]){
      Matrix A = newMatrix(10);
        changeEntry(A, 2, 1, 2);
        changeEntry(A, 3, 1, 5);
        changeEntry(A, 1, 2, 2);
        changeEntry(A, 1, 3, 5);
        changeEntry(A, 1, 1, 4);
        changeEntry(A, 2, 2, 2);
        changeEntry(A, 2, 5, 0);
        changeEntry(A, 2, 3, 0);
        changeEntry(A, 3, 3, 5);
        if (getNNZ(A) != 7) return 1;
        changeEntry(A, 1, 3, 0);
        changeEntry(A, 3, 1, 0);
        changeEntry(A, 3, 3, 0);
        if (getNNZ(A) != 4) return 2;
        changeEntry(A, 7, 6, 42);
        changeEntry(A, 10, 1, 24);
        if (getNNZ(A) != 6) return 3;
        changeEntry(A, 7, 6, 0);
        if (getNNZ(A) != 5) return 4;
        makeZero(A);
        changeEntry(A, 5, 5, 5);
        if (getNNZ(A) != 1) return 5;
}